app/src/main/java/com/github/sdp/mediato/ui/NewItemFragment.java

```diff
@@ -31,51 +33,69 @@
 import com.google.gson.Gson;
 
 import java.io.IOException;
-import java.util.Locale;
 
 public class NewItemFragment extends Fragment {
-    // The maximum allowed length for review field
-    private OLAPI oLAPI = new OLAPI("https://openlibrary.org/");
-
     public final static int MAX_REVIEW_LENGTH = 100;
     public final static int MAX_SUMMARY_LENGTH = 300;
-
     public WebView webView;
+    public EditText reviewText;
+    // The maximum allowed length for review field
+    private OLAPI oLAPI = new OLAPI("https://openlibrary.org/");
     private View view;
     private Media media;
     private FragmentSwitcher fragmentSwitcher;
 
     @Nullable
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
-
         View view = inflater.inflate(R.layout.fragment_new_item, container, false);
         this.view = view;
 
-        EditText review = view.findViewById(R.id.item_review_edittext);
+        reviewText = view.findViewById(R.id.item_review_edittext);
+        InputFilter[] filters = new InputFilter[] { new InputFilter.LengthFilter(MAX_REVIEW_LENGTH) };
+        reviewText.setFilters(filters);
+
         fragmentSwitcher = (FragmentSwitcher) getActivity();
 
         Button addButton = view.findViewById(R.id.item_button_add);
-        addButton.setOnClickListener(v -> addItem(view));
+        addButton.setOnClickListener(v -> addItem(view, reviewText));
 
         Bundle bundle = this.getArguments();
         if (bundle != null) {
             media = (Media) bundle.get("media");
         }
 
         setItemInformation(media.getTitle(), media.getSummary(), media.getPosterUrl());
-
+        setObserverOnKeyboard(view);
         loadDescriptionIfMissing();
-
         setProgressBarIndicator();
-
         searchTrailer();
 
         webView = view.findViewById(R.id.trailer_web_view);
 
         return view;
     }
 
+    private void setObserverOnKeyboard(View view) {
+        // Get references to the views
+        ScrollView scrollView = view.findViewById(R.id.scrollView);
+
+        // Set an OnLayoutChangeListener to the root view to detect layout changes
+        scrollView.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+            @Override
+            public void onGlobalLayout() {
```

Comment:
I'm not sure I understand what these calculations are for. Can you explain ?

Reply:
it just to detect when the keyboard open and scroll down to the bottom to see the review text and the button add